SU2_CFD/src/solvers/CNEMOEulerSolver.cpp

```diff
@@ -0,0 +1,3172 @@
+    nodes      = new CNEMONSVariable    (Pressure_Inf, MassFrac_Inf, Mvec_Inf,
+                                         Temperature_Inf, Temperature_ve_Inf,
+                                         nPoint, nDim, nVar, nPrimVar, nPrimVarGrad,
+                                         config, FluidModel);
+    node_infty = new CNEMONSVariable    (Pressure_Inf, MassFrac_Inf, Mvec_Inf,
+                                        Temperature_Inf, Temperature_ve_Inf,
+                                        1, nDim, nVar, nPrimVar, nPrimVarGrad,
+                                        config, FluidModel);    
+  } else {    
+    nodes      = new CNEMOEulerVariable(Pressure_Inf, MassFrac_Inf, Mvec_Inf,
+                                        Temperature_Inf, Temperature_ve_Inf,
+                                        nPoint, nDim, nVar, nPrimVar, nPrimVarGrad,
+                                        config, FluidModel);
+    node_infty = new CNEMOEulerVariable(Pressure_Inf, MassFrac_Inf, Mvec_Inf,
+                                        Temperature_Inf, Temperature_ve_Inf,
+                                        1, nDim, nVar, nPrimVar, nPrimVarGrad,
+                                        config, FluidModel);        
+  }
+  SetBaseClassPointerToNodes();
+
+  check_infty = node_infty->SetPrimVar(0, FluidModel);
+
+  /*--- Check that the initial solution is physical, report any non-physical nodes ---*/
+
+  counter_local = 0;
+  for (iPoint = 0; iPoint < nPoint; iPoint++) {
+    nonPhys = nodes->SetPrimVar(iPoint, FluidModel);
+
+    if (nonPhys) {
+
+      sqvel = 0.0;
+
+      /*--- Set mixture state ---*/
+      FluidModel->SetTDStatePTTv(Pressure_Inf, MassFrac_Inf, Temperature_Inf, Temperature_ve_Inf);
+
+      /*--- Compute other freestream quantities ---*/
+      Density_Inf    = FluidModel->GetDensity();
+      Soundspeed_Inf = FluidModel->GetSoundSpeed();
+      for (iDim = 0; iDim < nDim; iDim++){
+        sqvel += Mvec_Inf[iDim]*Soundspeed_Inf * Mvec_Inf[iDim]*Soundspeed_Inf;
+      }      
+      Energies_Inf = FluidModel->GetMixtureEnergies();
+
+      /*--- Initialize Solution & Solution_Old vectors ---*/
+      for (iSpecies = 0; iSpecies < nSpecies; iSpecies++) {
+        Solution[iSpecies]      = Density_Inf*MassFrac_Inf[iSpecies];
+      }
+      for (iDim = 0; iDim < nDim; iDim++) {
+        Solution[nSpecies+iDim] = Density_Inf*Mvec_Inf[iDim]*Soundspeed_Inf;
+      }
+      Solution[nSpecies+nDim]     = Density_Inf*(Energies_Inf[0] + 0.5*sqvel);
+      Solution[nSpecies+nDim+1]   = Density_Inf*Energies_Inf[1];
+      nodes->SetSolution(iPoint,Solution);
+      nodes->SetSolution_Old(iPoint,Solution);
+      counter_local++;
+    }
+  }
+
+  /*--- Warning message about non-physical points ---*/
+  if (config->GetComm_Level() == COMM_FULL) {
+#ifdef HAVE_MPI
+    SU2_MPI::Reduce(&counter_local, &counter_global, 1, MPI_UNSIGNED_LONG, MPI_SUM, MASTER_NODE, MPI_COMM_WORLD);
+#else
+    counter_global = counter_local;
+#endif
```

Comment:
Glad to see the drop in lines of code :)
While I'm here, These type of #ifdef HAVE_MPI blocks are very rarely needed, SU2_MPI:: does the direct copy when the code is compiled without MPI.